cfme/cloud/tenant.py

```diff
@@ -68,10 +72,10 @@ def wait_for_disappear(self, timeout=300):
         except TimedOutError:
             logger.error('Timed out waiting for tenant to disappear, continuing')
 
-    # def wait_for(self, timeout=600):
-    #     return wait_for(self.exists, fail_condition=False, timeout=timeout,
-    #                     message='Wait for cloud tenant to appear', delay=10,
-    #                     fail_func=sel.refresh)
+    def wait_for_appear(self, timeout=600):
+        return wait_for(self.exists, fail_condition=False, timeout=timeout,
```

Comment:
**Required**
Drop `fail_condition` and `fail_func` parameters. fail_condition is False by default and self.exists calls navigate, which refreshes the page. This fail_func refresh would be executed right before the navigation that occurs during self.exists anyway.